### ISSUE TITLE
`proxy-agent` is not working with http proxy & web sockets 

### ISSUE THREAD

**Issue by OpportunityLiu** [TooTallNate/proxy-agents#176]
Since proxies will drop hop-by-hop headers (`Connect` and `Upgrade`), we cannot use `http-proxy-agent` to connect to a web socket. But when I use `proxy-agent`, it will send `ws:` requests via `http-proxy-agent`.

**Comment by TooTallNate** [comment]
Can you share your code?

**Comment by OpportunityLiu** [comment]
Here is a simple repo:

proxy.js
```js
import * as http from 'http';
import { createProxy } from 'proxy';

const server = createProxy(http.createServer());
server.listen(3128, () => {
    var port = server.address().port;
    console.log('HTTP(s) proxy server listening on port %d', port);
});
```

server.js
```js
import { createServer } from 'node:http';
import { WebSocketServer } from 'ws';

const s = createServer((req, res) => {
    console.log('request', req.socket.remoteAddress, req.headers);
    res.writeHead(200, { 'Content-Type': 'text/plain' });
    res.end('okay');
});
s.listen(8080);
const wss = new WebSocketServer({ server: s });
wss.on('listening', () => {
    console.log('Ws server listening');
});
wss.on('connection', (ws, req) => {
    console.log('connected', req.socket.remoteAddress, req.headers);
});
```

client.js
```js
import { WebSocket } from 'ws';
import { ProxyAgent } from 'proxy-agent';

const ws1 = new WebSocket('ws://localhost:8080');

process.env['http_proxy'] = 'http://localhost:3128';
process.env['https_proxy'] = 'http://localhost:3128';
process.env['no_proxy'] = '';
const agent = new ProxyAgent();
const ws2 = new WebSocket('ws://localhost:8080', { agent });
```

Start `server.js` and `proxy.js`, then run `client.js`.

In `server.js`, you will get:
```
Ws server listening
connected ::ffff:127.0.0.1 {
  'sec-websocket-version': '13',
  'sec-websocket-key': 'cfNz4mfeJergcs4XGTEQOg==',
  connection: 'Upgrade',
  upgrade: 'websocket',
  'sec-websocket-extensions': 'permessage-deflate; client_max_window_bits',
  host: 'localhost:8080'
}
request ::ffff:127.0.0.1 {
  'sec-websocket-version': '13',
  'sec-websocket-key': 'vjKV3Bqg0vHkNuvBS4CGQg==',
  'sec-websocket-extensions': 'permessage-deflate; client_max_window_bits',
  host: 'localhost:8080',
  'proxy-connection': 'close',
  'x-forwarded-for': '::ffff:127.0.0.1',
  via: '1.1 xxx (proxy/2.1.1)',
  connection: 'close'
}
```

In `client.js`, you will get:
```
node:events:491
      throw er; // Unhandled 'error' event
      ^

Error: Unexpected server response: 200
    at ClientRequest.<anonymous> (...\node_modules\ws\lib\websocket.js:888:7)
    at ClientRequest.emit (node:events:513:28)
    at HTTPParser.parserOnIncomingClient (node:_http_client:693:27)
    at HTTPParser.parserOnHeadersComplete (node:_http_common:128:17)
    at Socket.socketOnData (node:_http_client:534:22)
    at Socket.emit (node:events:513:28)
    at addChunk (node:internal/streams/readable:315:12)
    at readableAddChunk (node:internal/streams/readable:289:9)
    at Socket.Readable.push (node:internal/streams/readable:228:10)
    at TCP.onStreamRead (node:internal/stream_base_commons:190:23)
Emitted 'error' event on WebSocket instance at:
    at emitErrorAndClose (...\node_modules\ws\lib\websocket.js:1008:13)
    at processTicksAndRejections (node:internal/process/task_queues:83:21)
```

**Comment by OpportunityLiu** [comment]
Currently a workaround is always to use `https-proxy-agent` for http proxies.

```js
import { proxies } from 'proxy-agent';

proxies['http'][0] = proxies['http'][1];
proxies['https'][0] = proxies['https'][1];
```

**Comment by TooTallNate** [comment]
Fixed in [`proxy-agent@6.2.1`](https://github.com/TooTallNate/proxy-agents/releases/tag/proxy-agent%406.2.1).
